### ISSUE TITLE
Upgrade to HA Core 2024.12.3 breaks BA integration

### ISSUE THREAD

**Issue by MeeperBeeper** [dahlb/ha_blueair#162]
Well it was good while it lasted...

I upgraded to HA Core 2024.12.3 earlier today and it completely broke the Blueair integration.

Any attempt to click on pretty much anything resulted in no action.  In the logs I saw the following:
```
2024-12-13 12:19:46.032 ERROR (MainThread) [homeassistant.setup] Setup failed for custom integration 'ha_blueair': Unable to import component: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1014, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        self._get_component, True
        ^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1074, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1022, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/ha_blueair/__init__.py", line 16, in <module>
    from blueair_api import get_devices, get_aws_devices, LoginError
ImportError: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1034, in async_get_component
    self._component_future.result()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1026, in async_get_component
    comp = self._get_component()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1074, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1022, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/ha_blueair/__init__.py", line 16, in <module>
    from blueair_api import get_devices, get_aws_devices, LoginError
ImportError: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
```

After much fussing, I finally managed to force the integration's removal and (after it telling me "omg invalid config") the config that went with it hoping to completely remove everything and start over.

I was then able to re-add the integration via HACS but when I went to setup my account/device again, I got the following error on the screen:

`Config flow could not be loaded: {"message":"Invalid handler specified"}`

Checking the pretty log I got this:
```
Logger: homeassistant.config_entries
Source: config_entries.py:3314
First occurred: 1:02:18 AM (2 occurrences)
Last logged: 1:04:53 AM

Error occurred loading flow for integration ha_blueair: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
```

With the more detailed log showing this:
```
2024-12-13 17:02:18.562 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to import_module with args ('custom_components.ha_blueair.config_flow',) inside the event loop by integration 'config' at homeassistant/components/config/config_entries.py, line 179: return await super()._post_impl(request, data) (offender: /usr/src/homeassistant/homeassistant/loader.py, line 1303: return importlib.import_module(f"{self.pkg_path}.{platform_name}")), please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+config%22
For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#import_module
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 227, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 213, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 154, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 708, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 679, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 2027, in _run_once
    handle._run()
  File "/usr/local/lib/python3.13/asyncio/events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.13/site-packages/aiohttp/web_protocol.py", line 480, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.13/site-packages/aiohttp/web_app.py", line 569, in _handle
    return await handler(request)
  File "/usr/local/lib/python3.13/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 210, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 86, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 74, in wrapper
    return await method(view, request, data, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 172, in post
    return await self._post_impl(request, data)
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 179, in _post_impl
    return await super()._post_impl(request, data)
2024-12-13 17:02:18.564 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration ha_blueair: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
2024-12-13 17:04:53.136 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration ha_blueair: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
2024-12-13 17:04:53.136 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration ha_blueair: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
2024-12-13 17:04:53.136 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration ha_blueair: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
2024-12-13 17:04:53.136 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration ha_blueair: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
2024-12-13 17:04:53.136 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration ha_blueair: cannot import name 'get_devices' from 'blueair_api' (/usr/local/lib/python3.13/site-packages/blueair_api/__init__.py)
```

I'm guessing the HA group has changed something in the latest version which has now broken how the BA integration module works.

I'm open to suggestions, or even finding out if someone else has managed to upgrade to HA Core 2024.12.3 and not had this issue.

**Comment by rainwoodman** [comment]
I am running 12.3 and latest ha_blueair release with no apparent issue. I suspect somehow your system is stuck with an older version of blueair_api -- did you manually install blueair_api on your ha supervisor(I think that's what it is called?) at any point?

**Comment by MeeperBeeper** [comment]
Hey @rainwoodman, 

Thanks for the confirmation that I'm crazy (aka - it's just me that's broken).

No, I hadn't manually installed the blueair_api pip package or even done anything involving the CLI in HA yet.

After multiple more deletes/adds of the addon through HACS, I managed to get into the CLI of the HA Core docker container (I think, still not sure of the HA architecture).  Once there I found the blueair_api package and further found that uninstalling the addon does not remove the blueair_api python package this addon depends on.

Once that far into the HA system, I managed to use pip to remove the blueair_api package.  Once I did that and reinstalled this addon (which re-added this pip package) then everything started working again.

Not sure what I did that necessitated such drastic steps but at last I managed to get it fixed.

Thanks again for the confirmation that it was just me.
